src/Validator/MaxFiscalYearDonationValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace AppBundle\Validator;
+
+use AppBundle\Donation\DonationRequest;
+use AppBundle\Repository\TransactionRepository;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class MaxFiscalYearDonationValidator extends ConstraintValidator
+{
+    private $transactionRepository;
+
+    public function __construct(
+        TransactionRepository $transactionRepository
+    ) {
+        $this->transactionRepository = $transactionRepository;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof MaxFiscalYearDonation) {
+            throw new UnexpectedTypeException($constraint, MaxFiscalYearDonation::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        /** @var DonationRequest $donationRequest */
+        if (!($donationRequest = $this->context->getObject()) instanceof DonationRequest) {
+            throw new UnexpectedTypeException($value, DonationRequest::class);
+        }
+
+        if (!$email = $donationRequest->getEmailAddress()) {
+            return;
+        }
+
+        $totalCurrentAmount = $this->transactionRepository->getTotalAmountCurrentYearByEmail($email);
```

Comment:
If $value > limit no need to check the DB

Reply:
Yes, but how do you get total_current_amount in the message ? ^^,